Sources/Kafka/Configuration/KafkaConfiguration+Metrics.swift

```diff
@@ -0,0 +1,133 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the swift-kafka-client open source project
+//
+// Copyright (c) 2023 Apple Inc. and the swift-kafka-client project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of swift-kafka-client project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Metrics
+
+extension KafkaConfiguration {
+    // MARK: - Metrics
+
+    /// Use to configure metrics.
+    public struct KafkaMetrics: Sendable {
+        internal var someMetricsSet: Bool {
+            self.timestamp != nil ||
+                self.time != nil ||
+                self.age != nil ||
+                self.replyQueue != nil ||
+                self.messageCount != nil ||
+                self.messageSize != nil ||
+                self.messageMax != nil ||
+                self.messageSizeMax != nil ||
+                self.totalRequestsSent != nil ||
+                self.totalBytesSent != nil ||
+                self.totalResponsesRecieved != nil ||
+                self.totalBytesReceived != nil ||
+                self.totalMessagesSent != nil ||
+                self.totalMessagesBytesSent != nil ||
+                self.totalBytesReceived != nil ||
+                self.metadataCacheCount != nil
+        }
+
+        /// librdkafka's internal monotonic clock (microseconds)
+        public var timestamp: Gauge?
+        /// Wall clock time in seconds since the epoch
+        public var time: Timer?
+        /// Time since this client instance was created
+        public var age: Timer?
+        /// Number of ops (callbacks, events, etc) waiting in queue for application to serve
+        public var replyQueue: Gauge?
+        /// Current number of messages in producer queues
+        public var messageCount: Gauge?
+        /// Current total size of messages in producer queues
+        public var messageSize: Gauge?
```

Comment:
This is producer specific should we only make this available on the producer configuration?

```suggestion
        /// Current number of queued producer messages.
        public var queuedProducerMessages: Gauge?
        /// Current total size in bytes of queued producer messages.
        public var queuedProducerMessagesSize: Gauge?
```

Reply:
Yep, I think we can

Reply:
I've renamed it but do you think we should make `ProducerMetrics` and `ConsumerMetrics` and allow only specific ones?